src/main/java/uk/gov/pay/directdebit/events/model/GoCardlessMandateAction.java

```diff
@@ -0,0 +1,20 @@
+package uk.gov.pay.directdebit.events.model;
+
+import java.util.Optional;
+
+/**
+ * GoCardless mandate actions
+ *
+ * @see <a href="https://developer.gocardless.com/api-reference/#events-mandate-actions">https://developer.gocardless.com/api-reference/#events-mandate-actions</a>
+ */
+public enum GoCardlessMandateAction {
```

Comment:
I’m not really sure this enum needs to exist at all, really. We don’t use it with the state update stuff when handling events; we just use strings.

If this enum does continue to exist, I think it should represent all the possible GoCardless mandate actions, rather than just the ones we want to “handle”. We don’t really have a unified idea of handling an event now that we have the state updates separated from the sending of emails and logging of events.

Reply:
| If this enum does continue to exist, I think it should represent all the possible GoCardless mandate actions, rather than just the ones we want to “handle”

Maybe we can rename this to "RelevantGoCardlessMandateActions"?

Reply:
I guess what I was getting at is that I don’t think we necessarily need a global list of event actions we care about. Essentially we have:

- events that trigger mandate/payment state updates
- events that send emails
- events that log an error
- maybe other things in the future

… and these can overlap (an event could trigger a state update and send an email). Importantly, there are events that fit in to none of these categories and do nothing (they just get stored).

I’m not even really sure we need `GoCardlessMandateHandler` and friends. We could just do something like this:

```java
events.forEach(goCardlessService::storeEvent);
updateStatesForEvents(events);
sendEmailsForEvents(events);
logErrorsForEvents(events);
```

Each method (which could be in a separate service to keep `WebhookGoCardlessService` light) would do its own thing to decide what events are relevant and what it wants to do with them.

On the specifics of the `GoCardlessMandateAction` enum: if we have it, I think it should represent every possible mandate action, so that we could also use it as a field in `GoCardlessEvent` etc. So I’d be in favour of having a more specific name for what we currently have.

Reply:
ok. I've created a GoCardlessEventHandler class. it looks much nicer and I managed to achieve a net deletion of code!